src/Akka.Streams.Kafka/Messages/MessageAndMeta.cs

```diff
@@ -0,0 +1,16 @@
+using Confluent.Kafka;
+
+namespace Akka.Streams.Kafka.Messages
+{
+    /// <summary>
+    /// Container for storing message and topic/parition info
+    /// </summary>
+    /// <typeparam name="K">Type of key</typeparam>
+    /// <typeparam name="V">Type of value</typeparam>
+    public class MessageAndMeta<K, V>
```

Comment:
Which parts of the solution are going to use this message type?

Reply:
Previously, this message type was build-in in Confluence.Kafka, and it was used to store message and destination (topic or topic partition info) before passing to kafka producer. So this is exclusively for Producer part.
Now new api require to pass topic or topic partition as a separate parameter to Produce method, along with message data. But it is just a matter of presentation, I think having a stream of `MessageAndMetadata`  objects is most simple way to pass messages and metadata.

So now I am doing the following to use required overload of Produce method:

``` csharp
onPush: () =>
{
    var msg = Grab(_stage.In);
    var result = new TaskCompletionSource<DeliveryReport<K, V>>();

    void PublishAction(Action<DeliveryReport<K, V>> report)
    {
        if (msg.TopicPartition != null)
            _producer.Produce(msg.TopicPartition, msg.Message, report);
        else
            _producer.Produce(msg.Topic, msg.Message, report);
    }

    PublishAction(report =>
    {
```

Reply:
Got it, that makes sense